### PR TITLE
Avoid the use of 'for (int' constructions ##build

### PR DESCRIPTION
**Pull request by radare**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [radare2 book](https://github.com/radareorg/radare2book) with the relevant information (if needed)

**Detailed description**

without this r2 doesnt build on some compilers

**Test plan**

see r2frida travis or try any random old bsd

**Closing issues**

nope

### CODE REVIEW

**Comment by ret2libc** [comment]
Can we wait a little bit and I try to push these changes to tree sitter upstream? I'd like to avoid changes to external repositories

**Comment by XVilka** [comment]
What about merging without tree-sitter changes?

**Comment by XVilka** [comment]
```
test_list.c:388:6: error: redefinition of 'i'
        int i;
            ^
test_list.c:380:6: note: previous definition is here
        int i;
            ^
test_list.c:416:6: error: redefinition of 'i'
        int i;
            ^
test_list.c:408:6: note: previous definition is here
        int i;
            ^
2 errors generated.
```

**Comment by radare** [comment]
Imho we cant wait. Ive been receiving complains in DM for 3 days. Dont know how fast are treesitter people in merging and releasing, but i dont see a problem in having an early fix. Ill fix that compilation error

**Comment by XVilka** [comment]
@radare they are fast and PR is already there: https://github.com/tree-sitter/tree-sitter/pull/604

**Comment by ret2libc** [comment]
I think there is an easy workaround for those who need to compile things right now. CFLAGS=-std=c99 ./configure; make . Also, instead of DM we would like to see issues opened on GitHub with full details. Thanks

File: shlr/tree-sitter/lib/src/stack.c [hunk 1]
```diff
@@ -171,7 +171,8 @@ static bool stack__subtree_is_equivalent(Subtree left, Subtree right) {
 static void stack_node_add_link(StackNode *self, StackLink link, SubtreePool *subtree_pool) {
   if (link.node == self) return;
 
-  for (int i = 0; i < self->link_count; i++) {
+  int i;
```
Comment (ret2libc) [inline shlr/tree-sitter/lib/src/stack.c:174]: didn't you say that your PR was not touching tree-sitter?

Comment (radare) [reply shlr/tree-sitter/lib/src/stack.c:174]: nope, i didnt said this

**Comment by radare** [comment]
i said we should fix the build until we can update treesitter when their code is compilable

File: shlr/tree-sitter/lib/src/stack.c (hunk 1 above)
Comment (ret2libc) [reply shlr/tree-sitter/lib/src/stack.c:174]: this is not needed. remove it please. tree-sitter is already compiled with `-std=c99` so it does not complain. https://github.com/radareorg/radare2/blob/master/shlr/Makefile#L334

**Comment by radare** [comment]
done

**Pull request merged**